common/odigos_config.go

```diff
@@ -1,5 +1,241 @@
+	}
+
+}
+
+type ConfigCollectorNode CollectorNodeConfiguration
+
+func (data *ConfigCollectorNode) ToString() {
+	if data == nil {
+		fmt.Println(": not set")
+		return
+	} else {
+		var placeholder ConfigString = ConfigString(data.K8sNodeLogsDirectory)
+		placeholder.ToString()
+	}
+}
+
+// go doesn't allow type to deal with pointers, so it is wrapping around MountMethod
+type ConfigMountMethod MountMethod
+
+func (data *ConfigMountMethod) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		fmt.Printf(": %v\n", *data)
+	}
+}
+
+// could possibly delete
+type ConfigUIMode UiMode
+
+func (data ConfigUIMode) ToString() {
+	fmt.Printf(": %v\n", data)
+}
+
+type ConfigEnvInjection EnvInjectionMethod
+
+func (data *ConfigEnvInjection) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		fmt.Printf(": %v\n", *data)
+	}
+}
+
+type ConfigNodeSelector map[string]string
+
+func (data ConfigNodeSelector) ToString() {
+	if len(data) == 0 {
+		fmt.Printf(": not set\n")
+	} else {
+		fmt.Printf(": \n")
+		for key, value := range data {
+			fmt.Printf("key: %+v, value: %+v\n", key, value)
+		}
+	}
+}
+
+type ConfigUserEnv UserInstrumentationEnvs
+
+func (data *ConfigUserEnv) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else if len(data.Languages) == 0 {
+		fmt.Printf(": not set\n")
+	} else {
+		fmt.Printf(": \n")
+		for key, value := range data.Languages {
+			fmt.Printf("language: %+v, mode: %+v\n", key, value)
+		}
+	}
+}
+
+type ConfigRollout RolloutConfiguration
+
+func (data *ConfigRollout) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		placeholder := ConfigBoolPointer{Value: data.AutomaticRolloutDisabled}
+		placeholder.ToString()
+	}
+}
+
+type ConfigCollectorGateway CollectorGatewayConfiguration
+
+func (data *ConfigCollectorGateway) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		placeholder := ConfigBoolPointer{Value: data.ServiceGraphDisabled}
+		placeholder.ToString()
+	}
+}
+
+type ConfigOidcTenant OidcConfiguration
+
+func (data *ConfigOidcTenant) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		var placeholder ConfigString = ConfigString(data.TenantUrl)
+		placeholder.ToString()
+	}
+}
+
+type ConfigOidcClientId OidcConfiguration
+
+func (data *ConfigOidcClientId) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		var placeholder ConfigString = ConfigString(data.ClientId)
+		placeholder.ToString()
+	}
+}
+
+type ConfigOidcClientSecret OidcConfiguration
+
+func (data *ConfigOidcClientSecret) ToString() {
+	if data == nil {
+		fmt.Printf(": not set\n")
+	} else {
+		var placeholder ConfigString = ConfigString(data.ClientSecret)
+		placeholder.ToString()
+	}
+}
+
+func makeAMap(config *OdigosConfiguration) map[string]ConfigField {
+	displayData := map[string]ConfigField{
+		consts.TelemetryEnabledProperty:           ConfigBool(config.TelemetryEnabled),
```

Comment:
instead of doing all these casts, update the actual types used in the OdigosConfig to the new redefined types. For our own custom structs, you can just add the ToString method directly to those types, so you don't even need to change them in the OdigosConfig (just add the function)